includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -111,16 +122,146 @@ public function scheduled_subscription_payment( $amount, $renewal_order ) {
 	 * @param WC_Order        $renewal_order The failed renewal order.
 	 */
 	public function update_failing_payment_method( $subscription, $renewal_order ) {
-		$renewal_order_tokens = $renewal_order->get_payment_tokens();
-		$renewal_token_id     = end( $renewal_order_tokens );
-		$renewal_token        = ! $renewal_token_id ? null : WC_Payment_Tokens::get( $renewal_token_id );
+		$renewal_token = $this->get_active_token_from_order( $renewal_order );
 		if ( is_null( $renewal_token ) ) {
 			Logger::error( 'Failing subscription could not be updated: there is no saved payment token for order #' . $renewal_order->get_id() );
 			return;
 		}
 		$this->add_token_to_order( $subscription, $renewal_token );
 	}
 
+	/**
+	 * Include the payment meta data required to process automatic recurring payments so that store managers can
+	 * manually set up automatic recurring payments for a customer via the Edit Subscriptions screen in 2.0+.
+	 *
+	 * @param array           $payment_meta Associative array of meta data required for automatic payments.
+	 * @param WC_Subscription $subscription The subscription order.
+	 * @return array
+	 */
+	public function add_subscription_payment_meta( $payment_meta, $subscription ) {
+		$active_token = $this->get_active_token_from_order( $subscription );
+
+		$payment_meta[ WC_Payment_Gateway_WCPay::GATEWAY_ID ] = [
+			self::PAYMENT_METHOD_META_TABLE => [
+				self::PAYMENT_METHOD_META_KEY => [
+					'label' => __( 'Saved payment method', 'woocommerce-payments' ),
+					'value' => empty( $active_token ) ? '' : strval( $active_token->get_id() ),
+				],
+			],
+		];
+
+		// Display select element on newer Subscriptions versions.
+		add_filter(
+			sprintf(
+				'woocommerce_subscription_payment_meta_input_%s_%s_%s',
+				WC_Payment_Gateway_WCPay::GATEWAY_ID,
+				self::PAYMENT_METHOD_META_TABLE,
+				self::PAYMENT_METHOD_META_KEY
+			),
+			function ( $input, $field_id, $field_value, $field_disabled ) use ( $subscription ) {
```

Comment:
The filter should provide `$subscription` as a parameter instead.

Reply:
I like this, thanks for the suggestion. Fixed in https://github.com/Automattic/woocommerce-payments/pull/892/commits/81fb6765eaa092812055effe11a6d286680ce1dd.